File: Chapter 05/JS/section_5_4_server_monitor/after/infrastructure/externalLibs/fakeLibs.js
/**
 * Dummies to allow execution without installing real npm packages.
 * This represents the "Chaotic Outside World".
 */
class TwilioClient {
    constructor(key) {
        this.key = key;
    }
    sendSms(to, msg) {
        // Simulation of a network call
    }
}

class FakeKafkaProducer {
    produce(topic, value) {
        console.log(`[Kafka] Pushed to ${topic}: ${value}`);
    }
}

module.exports = { TwilioClient, FakeKafkaProducer };

File: Chapter 05/C#/Section_5_4_Server_Monitor/After/Infrastructure/ExternalLibs/FakeLibs.cs
using System;

namespace Chapter05.ServerMonitor.After.Infrastructure.ExternalLibs
{
    /// <summary>
    /// Dummies to allow compilation without installing real NuGet packages.
    /// Represents the "Chaotic Outside World" SDKs.
    /// </summary>
    public class TwilioClient
    {
        private readonly string _key;

        public TwilioClient(string key) { _key = key; }

        public void SendSms(string to, string msg) 
        {
            // Accessing _key prevents the "mark as static" warning.
            _ = _key; 
        }
    }

    /// <summary>
    /// Interface for a generic message producer.
    /// Uses 'in' for contravariance to maximize architectural flexibility.
    /// </summary>
    public interface IProducer<in TKey, in TValue>
    {
        void Produce(TKey key, string topic, TValue value);
    }
}